### PR TITLE
components/form: add company select widget for 4th step registration challenge

### PR DESCRIPTION
**Pull request by maceksimon**
Add `FormFieldCompanySelect` and use it in `RegisterChallengePage`
Offers search functionality over an `QOptionGroup` component.

* Add tests
* Add translations

To do in next PR: Add dialog for registering a new company.

### CODE REVIEW

File: src/components/form/FormFieldCompanySelect.vue [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      },
+      {
+        label: 'Company 7',
+        value: 'company-7',
+      },
+    ]);
+
+    const filteredOptions = computed(() => {
+      return options.value.filter(
+        (option) =>
```
Comment (tmszi) [inline src/components/form/FormFieldCompanySelect.vue:79]: Missing type?

File: src/components/form/FormFieldCompanySelect.vue [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  name: 'FormFieldCompanySelect',
+  props: {
+    modelValue: {
+      type: String,
+      required: true,
+    },
+  },
+  emits: ['update:modelValue'],
+  setup(props, { emit }) {
+    const query = ref('');
```
Comment (tmszi) [inline src/components/form/FormFieldCompanySelect.vue:44]: Missing type?

File: src/components/form/FormFieldCompanySelect.vue [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    const inputValue = computed({
+      get(): string {
+        return props.modelValue;
+      },
+      set(value: string) {
+        emit('update:modelValue', value);
+      },
+    });
+
+    const borderRadius = rideToWorkByBikeConfig.borderRadiusCardSmall;
```
Comment (tmszi) [inline src/components/form/FormFieldCompanySelect.vue:95]: Missing type?

File: src/components/form/FormFieldCompanySelect.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/form/FormFieldCompanySelect.vue]: Added.

File: src/components/form/FormFieldCompanySelect.vue (hunk 3 above)
Comment (maceksimon) [reply src/components/form/FormFieldCompanySelect.vue]: Added.

**Review by tmszi — APPROVED** [review]

**Pull request merged**